extensions/serverless/functions.yaml

```diff
@@ -199,12 +225,44 @@ data:
           $language = 'nodejs' ? ($exists($root.spec.runtime) and $root.spec.runtime != 'python39') ? $root.spec.runtime : 'nodejs16' :
           $language = 'python' ? 'python39' :
           ''
-    - path: spec.runtimeImageOverride
+    - path: spec.resourceConfiguration.function.profile
+      type: string
+      simple: true
+      trigger: [profile]
+      enum: ['XS', 'S', 'M', 'L', 'XL']
+      subscribe:
+        changeResources: |-
+          $root.spec.resourceConfiguration.function.resources.limits.cpu = '100m' and 
+          $root.spec.resourceConfiguration.function.resources.limits.memory = '128Mi' and
+          $root.spec.resourceConfiguration.function.resources.requests.cpu = '50m' and
+          $root.spec.resourceConfiguration.function.resources.requests.memory = '64Mi' ? 'XS' :
+
+          $root.spec.resourceConfiguration.function.resources.limits.cpu = '200m' and 
+          $root.spec.resourceConfiguration.function.resources.limits.memory = '256Mi' and
+          $root.spec.resourceConfiguration.function.resources.requests.cpu = '100m' and
+          $root.spec.resourceConfiguration.function.resources.requests.memory = '128Mi' ? 'S' :
+
+          $root.spec.resourceConfiguration.function.resources.limits.cpu = '400m' and 
+          $root.spec.resourceConfiguration.function.resources.limits.memory = '512Mi' and
+          $root.spec.resourceConfiguration.function.resources.requests.cpu = '200m' and
+          $root.spec.resourceConfiguration.function.resources.requests.memory = '256Mi' ? 'M' :
+
+          $root.spec.resourceConfiguration.function.resources.limits.cpu = '800m' and 
+          $root.spec.resourceConfiguration.function.resources.limits.memory = '1024Mi' and
+          $root.spec.resourceConfiguration.function.resources.requests.cpu = '400m' and
+          $root.spec.resourceConfiguration.function.resources.requests.memory = '512Mi' ? 'L' :
+
+          $root.spec.resourceConfiguration.function.resources.limits.cpu = '1600m' and 
+          $root.spec.resourceConfiguration.function.resources.limits.memory = '2048Mi' and
+          $root.spec.resourceConfiguration.function.resources.requests.cpu = '800m' and
+          $root.spec.resourceConfiguration.function.resources.requests.memory = '1024Mi' ? 'XL' : ''
```

Comment:
probably here should be another option when limits/request are not matching XS, S, etc..., because right now I can't edit for example cpu request or when I make it very fast it makes a flip :D